renderer/canvas.py

```diff
@@ -653,20 +658,62 @@ def _draw_container(self, surface, obs):
             Draws the containers to surface
         """
         station_container_offset = self.config["container"]["constants"]["STATION_CONTAINER_OFFSET"]
-
         for literal, is_true in obs.predicates.items():
             if is_true and literal.name == "container_at":
                 container = literal.params[0].name
                 station = literal.params[1].name
                 container_pos = self._get_station_position(station)
-                container_pos[1] -= station_container_offset
+                name, _ = trim_item_ID(container)
+                container_pos[1] -= self.config["container"]["entities"][name]["constants"].get("STATION_CONTAINER_OFFSET", station_container_offset)
                 self._draw_container_image(surface, container, obs, container_pos * self.pix_square_size)
             if is_true and literal.name == "has_container":
                 container = literal.params[1].name
                 player = literal.params[0].name
                 container_pos = self.player_pose[player]["position"]
                 self._draw_container_image(surface, container, obs, container_pos * self.pix_square_size)
     
+    def _add_platforms_underneath_stations(self, stations, abstract_tile_matrix):
+        """
```

Comment:
always include a short description explaining what this function does before the longer description that provides extra context.